replace fast-deep-equal with fast-equals

themre opened the pull request:
Hi,
I quickly tested replacing fast-deep-equal with fast-equals and it proved to be a nice replacement. It also doesn't cause the error that i posted under https://github.com/puckeditor/puck/issues/1254. Hope we can continue with this to not cause others same issue.
Also great tool that you created,really nice API!

Closes: #1254

chrisvxd reviewed (changes requested):
Thanks for the contribution! Happy to evaluate this but your PR contains a lot of unrelated changes - formatting, import changes, etc.

Would you be able to strip it back just to the changes you're introducing, so I can test it properly?

themre commented:
hi @chrisvxd , thx for the reply. i reverted formatting changes (Biome kicked in, sorry). for the other changes (included CSS in demo, replaced `assert` with `with`), i had to do those since i was without styling when testing locally and locally build didn't work if i used `assert`. Hope this is ok.

themre commented on apps/demo/app/styles.css:
```diff
@@ -1,4 +1,5 @@
 @import url("https://rsms.me/inter/inter.css");
+@import "@/core/dist/index.css";
```
done

themre commented on packages/core/components/Puck/index.tsx:
```diff
@@ -1,41 +1,41 @@
+	useEffect,
+	useMemo,
+	useRef,
+	useState,
```
done
